src/flask_restalchemy/resources/resources.py

```diff
@@ -293,6 +293,7 @@ def delete(self, relation_id, id):
         related_obj = session.query(self._related_model).get(relation_id)
         collection = getattr(related_obj, self._relation_property.key)
         collection.remove(requested_obj)
+        session.delete(requested_obj)
```

Comment:
We had this behavior intentionally, for some reason I can't remember. I realise now that it wasn't the best decision, as it is a completely unexpected behavior.